ui/app/pages/swaps/awaiting-swap/awaiting-swap.js

```diff
@@ -105,8 +108,21 @@ export default function AwaitingSwap({
     category: 'swaps',
   });
 
-  const blockExplorerUrl =
-    txHash && getBlockExplorerUrlForTx({ chainId, hash: txHash }, rpcPrefs);
+  let blockExplorerUrl;
+  if (txHash && rpcPrefs.blockExplorerUrl) {
+    blockExplorerUrl = getBlockExplorerUrlForTx({ hash: txHash }, rpcPrefs);
```

Comment:
It looks like `getBlockExplorerUrlForTx` has its own means of generating URLs for custom networks, and it's substantially more error prone than the method used by `@metamask/etherscan-link`. But it should work for these two cases at least. We can improve this separately.

Reply:
Ticket created here: https://github.com/MetaMask/metamask-extension/issues/10749